src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -80,21 +80,37 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 var mangledName = obj["FunctionSignature"].Value<string>();
 
                 var description = libraryServices.GetFunctionDescriptor(mangledName);
-
-                if (type == typeof(DSVarArgFunction))
+                if (description == null)
```

Comment:
I'm not understanding this - why do we only create dummy nodes in the case that the description is null? - Or am I missing reading the code - I would assume we do it anytime the Type above is null on line 54.

Reply:
My testing is using a zero-touch node package. After uninstalling the package and loading my test graph I see that the type value is not null, even though the library is not available. Getting the description invokes the library services, which returns null because the library is not loaded. There may be a more direct way of making the query, but this call was fairly similar to the code that is called when loading XML

Reply:
If the type is not actually loaded into the app domain, it should be null when getType searches for it, it's possible you still have the type in memory but the library is not loaded.

Reply:
continued - my point is this is a case we need to handle where type is really null.

Reply:
@ColinDayOrg I think I understand whats happening:
zerotouch nodes share a base class which is always going to be loaded so that is why type is not null for this node in your package.

We need to test a nodeModel type which is null so that will be a non zero touch node.

Also, I'm still not sure a null description will work for zero touch node - couldn't it just be a zero touch node with no description? Also for zero touch nodes we'll need to handle the base class and the dsvararg class (variable inputs)

Also have you considered customNodes? (Function) class?

Reply:
After attempting to redo this check with something more specific (i.e. mirroring the code from the ZeroTouchNodeLoader) it became apparent that there needs to be some more infrastructure work to be able to utilize the new code. 

The two main reasons are;

1. In the XML format there is a field for zero-touch nodes that contains the assembly that the zero-touch node is in. This attribute is not currently present in the JSON format and will need to be added. This missing information is a blocking issue for updating this check to use the more specific code.
2. The name of the node has been moved to the view properties and is not available at the time the node model is loaded. This may not be an issue as the name appears to be used to identify built-in node types, and it is not clear if this is important at the point where we are loading zero-touch nodes.

I think that we may be able to use the more specific code once the assembly information is saved in the JSON format.

Reply:
@ColinDayOrg the assembly is saved as part of the concrete type like this `typname, assembly`

Reply:
I should have been more specific in my previous comment, I was talking about the path to the assembly, not the name of the assembly. 

In the case of the XML file format the assembly is saved as the path to the assembly; "C:\Users\dayc\AppData\Roaming\Dynamo\Dynamo Core\1.2\packages\DynamoPrimitives\bin\DynamoPrimitives.dll")

In the case of the JSON file format it is saved in a type name format; "Dynamo.Graph.Nodes.ZeroTouch.DSFunction").

Reply:
so we discussed this hintpath as part of another task, what code are you talking about it being required for? I can't remember the outcome of that discussion, I think we decided we could add hintpath as a view layer property since it will have no meaning outside of dynamo.

Reply:
The assembly information is used in ZeroTouchNodeLoader.cs, line 76-100, to retrieve the FunctionDescriptor for the given function. This appears to me to be the best way to determine if the zero-touch node can be resolved. Currently it seems that there is not enough information saved in the JSON format to do a formal retrieval of the needed information.

Reply:
I think we'll need to rethink this based on the other task. presently you make a guess based on the function descriptor being null?

Reply:
I know it's a bit of a style thing, but can you rename description to functionDescriptor or something like that so it's not confused with node description text.

Reply:
I have renamed the variable, I agree it is clearer. 

The choice to use the function descriptor being null as a switch was based on the code in ZeroTouchNodeLoader.cs where the same test is used to determine (when loading XML) if a dummy node should be created or not. Note though that the test in the zero-touch loader is slightly more complex, but uses information that is not currently available from the JSON format.